cmd/deploy/start_test.go

```diff
@@ -40,106 +43,135 @@ func (suite *DeployStartTestSuite) SetupTest() {
+			kind:        "kubernetes",
+			file:        testAppYamlStr,
+			contentType: "yaml",
+			expected: de.StartPipelineResponse{
+				PipelineID: "12345",
+			},
+			assertion: func(result []byte, expected de.StartPipelineResponse) error {
+				var received = FormattableDeployStartResponse{}
+				err := yaml.Unmarshal(result, &received)
+				if err != nil {
+					return err
+				}
+				suite.Equal(expected.PipelineID, received.DeploymentId, "they should be equal")
+				return nil
+			},
+		},
+		{
+			name:        "start pipeline yaml fail validation",
+			kind:        "kubernetes",
+			file:        testAppYamlStrInvalid,
+			contentType: "yaml",
+			expected: de.StartPipelineResponse{
+				PipelineID: "12345",
+			},
+			assertion: func(result []byte, expected de.StartPipelineResponse) error {
+				suite.Equal(expectedValidationError, string(result))
+				return nil
+			},
+		},
+		{
+			name:        "start pipeline yaml success lambda",
+			kind:        "lambda",
+			file:        testLambdaYamlStr,
+			contentType: "yaml",
+			expected: de.StartPipelineResponse{
+				PipelineID: "12345",
+			},
+			assertion: func(result []byte, expected de.StartPipelineResponse) error {
+				// verify there is no validation errors being shown to the user
+				o := string(result)
+				suite.NotContains(o, "YAML is NOT valid. See the following errors:")
+
+				// verify that response formats as expected
+				var received = FormattableDeployStartResponse{}
+				err := yaml.Unmarshal(result, &received)
+				if err != nil {
+					return err
+				}
+				suite.Equal(expected.PipelineID, received.DeploymentId, "they should be equal")
+				return nil
+			},
+		},
 	}
-	var received = FormattableDeployStartResponse{}
-	yaml.Unmarshal(output, &received)
-	suite.Equal(expected.PipelineID, received.DeploymentId, "they should be equal")
-}
 
-func (suite *DeployStartTestSuite) TestDeployStartYAMLFailValidation() {
-	expected := &de.StartPipelineResponse{
-		PipelineID: "12345",
-	}
-	err := registerResponder(expected, http.StatusAccepted)
-	if err != nil {
-		suite.T().Fatalf("TestDeployStartYAMLSuccess failed with: %s", err)
-	}
-	tempFile := util.TempAppFile("", "app", testAppYamlStrInvalid)
-	if tempFile == nil {
-		suite.T().Fatal("TestDeployStartYAMLSuccess failed with: Could not create temp app file.")
-	}
-	suite.T().Cleanup(func() { os.Remove(tempFile.Name()) })
-	outWriter := bytes.NewBufferString("")
-	cmd := getDeployCmdWithFileName(outWriter, tempFile.Name(), "yaml")
-	err = cmd.Execute()
-	if err != nil {
-		suite.T().Fatalf("TestDeployStartYAMLSuccess failed with: %s", err)
-	}
-	output, err := io.ReadAll(outWriter)
-	if err != nil {
-		suite.T().Fatalf("TestDeployStartYAMLSuccess failed with: %s", err)
+	for _, c := range cases {
+		suite.T().Run(c.name, func(t *testing.T) {
+			err := registerResponder(c.expected, http.StatusAccepted, c.kind)
+			if err != nil {
+				suite.T().Fatalf("%s failed with: %s", c.name, err)
```

Comment:
Will just give a brief summary instead of giving a full Frankie review. You're cleaning this test code up so you don't deserve my full madness 😄 
- Keep Going: https://github.com/golang/go/wiki/TestComments#keep-going. This should be `assert.NoError(t, err)`.
- use the `t` from the subtest function, not the global `suite.T()` - otherwise your errors will be attributed to the suite, not the subtest. 
- This pattern is generally ugly because you're mutating the `err` variable over and over:
```go
err := registerResponder(...)
if err != nil {
}
...

err = cmd.Execute()
if err != nil {
}
```

I think this is much better:
```go
if err := registerResponder(...); err != nil {
}
...

if err := cmd.Execute(); err != nil {
}
```

Each of these `err`s is scoped separately (you're not mutating it over and over again).

Reply:
🤦 yikes, this was a mess. I think I made it a bit better.